client/src/partials/cash/modals/transfer.modal.js

```diff
@@ -0,0 +1,113 @@
+angular.module('bhima.controllers')
+.controller('CashTransferModalController', CashTransferModalController);
+
+CashTransferModalController.$inject = [
+  '$uibModalInstance', '$q', 'SessionService', 'CurrencyService', 'VoucherService', 'CashboxService', 'AccountService', 'cashBox', 'uuid'
+];
+
+/**
+ * @module cash/modals/CashTransferModalController
+ *
+ * @description This controller is responsible transfering money between auxillary cash and a virement account
+*/
+
+function CashTransferModalController(ModalInstance, $q, sessionService, currencyService, voucherService, cashBoxService, accountService, cashBox, uuid) {
+  var vm = this; 
+
+  /** Attaching service to the scope **/
+  vm.cashBox = cashBox;
+  vm.currencyService = currencyService; 
+  vm.cashBoxService = cashBoxService;
+
+  /** check if the cashbox id is provided**/
+  vm.cashBoxIdMissed = !cashBox.id;
+
+  /**
+  * @function submit
+  * @param {boolean} invalid it is true if the form is valid
+  * @description for submitting a dialog content
+  **/
+  function submit (invalid){
+    if (invalid) { return; }
+    var voucher = processVoucher();
```

Comment:
@jniles 
After a second review, I noticed that I am just initialize data which is the controller task, may be the functions names are confused, but I am doing any thing special.
Move this code the server or client service will lead me to send it in this form : 
```
{ 
  uuid : 'some uuid',   /* this could also be empty! */
  currency_id : 1,

  /* ...  other voucher properties ... */

 items : [{
    uuid : 'some uuid',  /* this could also be empty! */
    debit : 12.2,
    credit : 0
  }, {
    uuid : 'some uuid',  /* this could also be empty! */
    debit : 0,
    credit : 12.2
  }]
}
```
Which is almost the same thing, just a initialization, I am processing nothing.